src/components/Contacts/ContactDetails/ContactDetailsTab/EditContactModal/EditContactModal.tsx

```diff
@@ -0,0 +1,696 @@
+                    <Select
+                      labelId="socail-type-label"
+                      value={account.type}
+                      fullWidth
+                      readOnly
+                    >
+                      <MenuItem value={account.type}>{account.name}</MenuItem>
+                    </Select>
+                  </FormControl>
+                  <ContactEditDeleteIconButton>
+                    <DeleteIcon />
+                  </ContactEditDeleteIconButton>
+                </Grid>
+              </Grid>
+            </ContactInputWrapper>
+          </>
+        ))}
+      </>
+    ) : (
+      <>
+        <ContactInputWrapper>
+          <ContactSocialIcon />
+          <Grid container spacing={3}>
+            <Grid item xs={6}>
+              <TextField label={t('Username/URL')} value={null} fullWidth />
+            </Grid>
+            <Grid item xs={6}>
+              <FormControl fullWidth>
+                <InputLabel id="social-type-label">{t('Type')}</InputLabel>
+                <Select labelId="socail-type-label" value={null} fullWidth>
+                  <MenuItem value={'facebook'}>{t('Facebook')}</MenuItem>
+                  <MenuItem value={'twitterk'}>{t('Twitter')}</MenuItem>
+                  <MenuItem value={'linkedin'}>{t('LinkedIn')}</MenuItem>
+                  <MenuItem value={'website'}>{t('Website')}</MenuItem>
+                </Select>
+              </FormControl>
+              <ContactEditDeleteIconButton>
+                <DeleteIcon />
+              </ContactEditDeleteIconButton>
+            </Grid>
+          </Grid>
+        </ContactInputWrapper>
+      </>
+    );
+  };
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      title={'Edit People'}
+      content={
+        <>
+          <ContactEditContainer>
+            <ContactInputWrapper>
+              <ContactInputField
+                label={t('Contact')}
+                value={contact.name}
+                fullWidth
+              />
+            </ContactInputWrapper>
+            <ContactInputWrapper>
+              <PrimaryContactIcon />
+
+              <FormControl fullWidth={true}>
+                <ContactPrimaryPersonSelectLabel id="primary-person-select-label">
+                  {t('Primary')}
+                </ContactPrimaryPersonSelectLabel>
+                <Select
+                  labelId="primary-person-select-label"
+                  value={contact.primaryPerson?.id}
+                  fullWidth={true}
+                >
+                  {contact.people.nodes.map((person) => {
+                    return (
+                      <MenuItem
+                        key={person.id}
+                        value={person.id}
+                      >{`${person.firstName} ${person.lastName}`}</MenuItem>
+                    );
+                  })}
+                </Select>
+              </FormControl>
+            </ContactInputWrapper>
+          </ContactEditContainer>
+          <Divider />
```

Comment:
What do you think about making the code cleaner, and more modular, by separating these sections into their own component files?

Reply:
What get's complicated about that is, I'm going to wrap this whole modal in a ```Formik``` component to handle form validation. This would then require me to have to pass in all of Formik's props like, ```handleChange```, ```setFieldValue```,  ```isSubmitting```, ```isValid```, ```errors```, ```touched```, etc to each of the components, along with any other necessary props. I would love to do it if I can find an easy way to pass all these props.

Reply:
I guess I could just pass in one formik prop object, that contains all these methods and values?

Reply:
Yeah that does make things difficult... Maybe a Formik object would work.  I hope that React would keep a reference to one object, and not make a copy the object for each section... I forget how React Props work when it comes to memory usage.

Reply:
Another thing I need to think about, and would appreciate input is, how to manage all the states of the inputs and then pass that into the mutation on form submission. I think formik may help with this, but I need to think a bit more on the best way structure/implement it.

Reply:
What does it look like to pass Formik around? I don't think we would want to store any state in these sub components but making them dumb and just holding UI could be cleaner.

Reply:
I'm going to go ahead and test it out locally and see how it is. I agree, the less state that has to be managed the better.